### PR TITLE
Downloads improvements

### LINKED ISSUE #635: Download choices
* Add more choices (at least full res jpeg)
* Sync the viewer download choices with show page (share partial)

### PR DESCRIPTION
**Pull request by jrochkind**
* full size JPG powered by riiif
* riiif_image_url helper
* live download links in viewer

Closes #635

### CODE REVIEW

File: app/views/curation_concerns/base/_chf_download_menu.html.erb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                  data: {
+                    content_hook: "dl-original-link",
+                    label: (member.id if member),
+                    # These action/labels are what sufia/hyrax uses, although
+                    # a bit weird. https://github.com/samvera/hyrax/blob/1e504c200fd9c39120f514ac33cd42cd843de9fa/app/assets/javascripts/hyrax/ga_events.js
+                    analytics_event: "Files",
+                    analytics_label: "Downloaded",
+                    analytics_value: (member.representative_id if member)
+                  }) do %>
+          Original Image
```
Comment (hackmastera) [inline app/views/curation_concerns/base/_chf_download_menu.html.erb]: Sort of weird to have one use `do` and the other not. Maybe make them both the same; I like the 2nd form for readability (first thing you see tells you what the link is for).

File: app/assets/javascripts/chf_image_viewer.js [hunk 2]
```diff
@@ -112,12 +112,14 @@ ChfImageViewer.prototype.selectThumb = function(thumbElement) {
...
-  var linkUrl   = thumbElement.getAttribute('data-member-show-url');
-  var tileSource = thumbElement.getAttribute('data-tile-source');
+  var id                    = thumbElement.getAttribute('data-member-id');
+  var index                 = thumbElement.getAttribute('data-index');
+  var shouldShowInfo        = thumbElement.getAttribute('data-member-should-show-info') == "true";
+  var title                 = thumbElement.getAttribute('data-title');
+  var linkUrl               = thumbElement.getAttribute('data-member-show-url');
+  var downloadOriginalUrl   = thumbElement.getAttribute('data-member-dl-original-url');
+  var downloadJpegUrl       = thumbElement.getAttribute('data-member-dl-jpeg-url');
+  var tileSource            = thumbElement.getAttribute('data-tile-source');
```
Comment (hackmastera) [inline app/assets/javascripts/chf_image_viewer.js]: I really dislike this type of alignment; it always ends up being another thing to maintain. If it's all the same to you I prefer a single space on either side of all `=`s.

File: app/assets/javascripts/chf_image_viewer.js [hunk 3]
```diff
@@ -127,6 +129,11 @@ ChfImageViewer.prototype.selectThumb = function(thumbElement) {
   document.querySelector('*[data-hook="viewer-navbar-info-link"]').href = linkUrl;
   document.getElementsByClassName('viewer-pagination-numerator').item(0).textContent = index;
 
+  var originalLink = document.querySelector('#chf-image-viewer *[data-content-hook=dl-original-link]');
```
Comment (hackmastera) [inline app/assets/javascripts/chf_image_viewer.js]: what does the `*` do here?

File: app/views/curation_concerns/base/_chf_download_menu.html.erb (hunk 1 above)
Comment (jrochkind) [inline app/views/curation_concerns/base/_chf_download_menu.html.erb]: Sorry, which one do you mean as the "second form"?  I usually only use `do` when there is embedded markup, this one must have used to have had embedded markup, but doesn't anymore, so yeah make sense without `do`.

File: app/assets/javascripts/chf_image_viewer.js (hunk 2 above)
Comment (jrochkind) [inline app/assets/javascripts/chf_image_viewer.js]: I love it myself. But okay!

File: app/assets/javascripts/chf_image_viewer.js (hunk 3 above)
Comment (jrochkind) [inline app/assets/javascripts/chf_image_viewer.js]: `*[data-content-hook=dl-original-link]` is CSS selector for "any tag at all ("*") with attribute `data-content-hook` with value `dl-original-link`"

**Pull request merged**